### ISSUE TITLE
Videos without a file extension do not work

### ISSUE THREAD

**Issue by JayWelsh** [biati-digital/glightbox#294]
**Describe the bug**
It doesn't open the Plyr instance if a video link is provided without an extension

**Are you able to reproduce the bug in the demo site**
Yes

**To Reproduce**
Use a URL such as this for an example: https://seenhaus.mypinata.cloud/ipfs/QmSs7gpv4NVPYmsR2twPbyGBowG4TD37cEqKa895txGjVz

There are no CORS issues, glightbox is obviously requiring an extension which doesn't make much sense, we should at least be able to specify the mime type.

**Expected behavior**
Pass the source to the Plyr video player

**Comment by sanderquirynen** [comment]
I have the same issue. My video ends in a version number, eg: `video.mp4?v=1` and doesn't show.
When i add _.mp4_ at the end it works. So for now I fixed it by making the url something like `video.mp4?v=1&ext=.mp4`

**Comment by frankie-tech** [comment]
I think adding a queryParam check onto the slide parsers regexs should be easy enough to do.

Or even just a function that removes them before checking with the regex could work too.

I might have time to fiddle with that this weekend. I'd say it's a good first issue though, so anyways should feel welcome to give it a shot 😊

**Comment by JayWelsh** [comment]
@frankie-tech in some cases it isn't viable to add query params to the end of the link, though. Sometimes, especially when loading content from IPFS, the link format ends with a hash of the file which is being referenced (e.g. https://seenhaus.mypinata.cloud/ipfs/QmSs7gpv4NVPYmsR2twPbyGBowG4TD37cEqKa895txGjVz) even though it could be a .mp4 file. Do you think there is any way to work around this?

**Comment by frankie-tech** [comment]
I don't think there is a *workaround* right now. Since the slide parser relies on that regex to pick the slide to use, it would need to end in that extension.

It should be possible to (eventually) add in support for a `data-*` attribute. If I were to pseudo-code, I imagine the API would probably end up with `data-video-src` attribute, letting you specify youtube, vimeo, or local.

```js
if (/* url regex */ || slideConfig.videoSrc === 'video-type') {
	/* do video stuff */
}
```

This could probably be done *fairly* reasonably, but it has more to do with making the API match. Since it's a holiday weekend, I might be able to fiddle around and make a *completely unofficial, will not support, best of luck, no guarantees* fork that attempts to support it.

**Comment by frankie-tech** [comment]
Here's what I came up with just on the fly: https://github.com/frankie-tech/glightbox/tree/video-type-attribute

It handles the "URL doesn't include a file type" issue, and nothing else. Since it's unofficial, don't expect any additions to be made to it. I would honestly fork it or copy the code to save your own copy, 'cause I am not particularly attached to this shoe-horned addition.

If you don't wanna use an unsupported/unofficial fork with an unknown life-span, you can also try the [`?queryParam` workaround](https://github.com/biati-digital/glightbox/issues/294#issuecomment-960968942).

I'll add a feature request tag, since down the road an implementation can probably be done that isn't so *forced*.

**Comment by biati-digital** [comment]
This should be fixed in 3.2.0. GLightbox will not check the video extension anymore, if it's not youtube or vimeo, we'll just handle it as a regular video.

You still need to define a data-type so GLightbox can process the URL correctly.

`href="https://seenhaus.mypinata.cloud/ipfs/QmSs7gpv4NVPYmsR2twPbyGBowG4TD37cEqKa895txGjVz" data-type="video"`

**Comment by JayWelsh** [comment]
Thanks so much @biati-digital !
